### PR TITLE
Add loss parameter to FeaturesEnricher

### PR DESCRIPTION
**Pull request by PoUpgini**

### CODE REVIEW

File: src/upgini/features_enricher.py [hunk 1]
```diff
@@ -198,6 +202,10 @@ def __init__(
         self.cv = cv
         if cv is not None:
             self.runtime_parameters.properties["cv_type"] = cv.name
+        self.loss = loss
+        if loss is not None:
+            self.loss = self.loss.lower()
+
```
Comment (c3p0-upgini) [inline src/upgini/features_enricher.py]: self.loss = loss.lower() if loss is not None else None

File: src/upgini/metrics.py [hunk 2]
```diff
@@ -72,13 +72,15 @@ def __init__(
         multiplier: int,
         cv: BaseCrossValidator,
         target_type: ModelTaskType,
+        add_params: Optional[Dict[str, Any]] = {},
```
Comment (c3p0-upgini) [inline src/upgini/metrics.py]: Лучше наверное дефолтом None делать

File: src/upgini/metrics.py [hunk 3]
```diff
@@ -396,6 +402,14 @@ def _get_cat_features(X: pd.DataFrame) -> List[str]:
     return [c for c in X.columns if not is_numeric_dtype(X[c])]
 
 
+def _get_add_params(input_params, add_params):
+    output_params = dict(input_params)
+    for k, v in add_params.items():
```
Comment (c3p0-upgini) [inline src/upgini/metrics.py]: Нет проверки, что add_params is None

Comment (c3p0-upgini) [reply src/upgini/metrics.py]: if add_params is not None:
   output_params.update(add_params)

File: src/upgini/utils/custom_loss_utils.py [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        if use_custom_loss:
+            runtime_parameters.properties["lightgbm_params_preselection.objective"] = loss
+            runtime_parameters.properties["lightgbm_params_base.objective"] = loss
+            runtime_parameters.properties["lightgbm_params_segment.objective"] = loss
+            msg = f"Using loss '{loss}' for feature selection"
+            logger.warning(msg)
+        else:
+            msg = f"Loss '{loss}' is not supported for feature selection with {model_task_type}"
+            logger.warning(msg)
```
Comment (c3p0-upgini) [inline src/upgini/utils/custom_loss_utils.py]: Если это сообщение нужно показать показать клиенту, то тогда помимо логирования нужно еще print(msg) делать и сам текст сообщения лучше вынести в файл strings.properties и подтягивать его через bundle.get("message_key").format(args)

File: src/upgini/utils/custom_loss_utils.py [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+                or (model_task_type == ModelTaskType.MULTICLASS)
+                and (loss in selection_loss_multi_clf)
+            )
+            else False
+        )
+
+        if use_custom_loss:
+            runtime_parameters.properties["lightgbm_params_preselection.objective"] = loss
+            runtime_parameters.properties["lightgbm_params_base.objective"] = loss
+            runtime_parameters.properties["lightgbm_params_segment.objective"] = loss
```
Comment (c3p0-upgini) [inline src/upgini/utils/custom_loss_utils.py:45]: А это какой-то задел на будущее, чтобы для разных частей можно было разные loss задавать?

Comment (PoUpgini) [reply src/upgini/utils/custom_loss_utils.py:45]: уже сейчас в MLBack три разных набора гиперпараметров под разные этапы отборв

**Review by c3p0-upgini — APPROVED** [review]
